src/utils/getScreenStructure.ts

```diff
@@ -11,7 +11,11 @@ const getScreenStructure = (
     return match([object, screen])
         .with(["contact", "home"], () => settings?.mapping_contact
             ? JSON.parse(settings.mapping_contact)
-            : STRUCTURE.CONTACT
+            : STRUCTURE.CONTACT.home
+        )
+        .with(["contact", "view"], () => settings?.mapping_contact
```

Comment:
Not sure how this `with` works. If it is `contact`, does it run all code blocks or just the first. Would an `if/else` be easier to follow with fewer dependancies?

Reply:
We use `ts-pattern` in almost all projects. `match` works as follows:

In `match`, we pass the value that we want to check. The value can be of any type (primitive, array, or object). In `.with`, we check for a match and return a value.

It's more like a `switch-case` but more powerful and flexible.

Reply:
It seems overkill and that vanilla typescript can be handled nicely with the return early pattern.

```typescript
    if (object === 'contact' && screen === 'home') {
        return settings?.mapping_contact
            ? JSON.parse(settings.mapping_contact)
            : STRUCTURE.CONTACT.home
    }
    if (object === 'contact' && screen === 'view') {
        return settings?.mapping_contact
            ? JSON.parse(settings.mapping_contact)
            : STRUCTURE.CONTACT.view
    }
    return [];
```

Or maybe even just
```typescript
    if (object === 'contact') {
        return settings?.mapping_contact
            ? JSON.parse(settings.mapping_contact)
            : (STRUCTURE.CONTACT[screen] ?? [])
    }
    return [];
```

Reply:
rewrote with first example